packages/react-hooks/src/hooks/debounced.ts

```diff
@@ -13,14 +13,16 @@ export function useDebouncedValue<T>(
   const [state, setState] = useState<T>(value);
 
   useEffect(() => {
-    const timeout = window.setTimeout(() => {
-      setState(value);
+    const timeout = setTimeout(() => {
+      if (value !== state) {
+        setState(value);
+      }
     }, timeoutMs);
 
     return () => {
-      window.clearTimeout(timeout);
+      clearTimeout(timeout);
     };
-  }, [value, timeoutMs]);
+  }, [value, state, timeoutMs]);
```

Comment:
You don't want to run this effect if the state change. We can probably use a ref instead.

Reply:
@tleunen guess we'll revert to the ref